router/router.go

```diff
@@ -1330,132 +1338,143 @@ func (rt *HandleT) generatorLoop() {
 	generatorStat := stats.NewTaggedStat("router_generator_loop", stats.TimerType, stats.Tags{"destType": rt.destName})
 	countStat := stats.NewTaggedStat("router_generator_events", stats.CountType, stats.Tags{"destType": rt.destName})
 
+	timeout := time.After(200 * time.Millisecond)
 	for {
-		generatorStat.Start()
-
-		if rt.guaranteeUserEventOrder {
-			//#JobOrder (See comment marked #JobOrder
-			rt.toClearFailJobIDMutex.Lock()
-			for idx := range rt.toClearFailJobIDMap {
-				wrk := rt.workers[idx]
-				wrk.failedJobIDMutex.Lock()
-				for _, userID := range rt.toClearFailJobIDMap[idx] {
-					delete(wrk.failedJobIDMap, userID)
+		select {
+		case pause := <-rt.generatorPauseChannel:
+			pkgLogger.Info("Generator loop is paused.")
+			rt.paused = true
+			pause.respChannel <- true
+			<-rt.generatorResumeChannel
```

Comment:
Let's add couple more logs here?
- Waiting to resume
- Resumed

Reply:
done